src/ray/object_manager/plasma/client.cc

```diff
@@ -235,11 +236,17 @@ class PlasmaClient::Impl : public std::enable_shared_from_this<PlasmaClient::Imp
   std::unordered_set<ObjectID> deletion_cache_;
   /// A mutex which protects this class.
   std::recursive_mutex client_mutex_;
+  /// Whether the current process should exit when read or write to the connection fails.
+  /// It should only be turned on when the plasma client is in a core worker.
+  bool exit_on_connection_failure_;
```

Comment:
Instead of modifying low-level `PlasmaClient`, why can't we reuse existing `check_signals_` mechanism in `CoreWorkerPlasmaStoreProvider`, which is the abstraction layer between the Core Worker and the plasma store? 

The `get` and `wait` paths in `CoreWorkerPlasmaStoreProvider` proactively check for interrupts via `check_signals_`. We can do the same for other Plasma operations.

`CoreWorker::ExitIfParentRayletDies()` already checks periodically if the raylet process is alive. Then let the CoreWorker take a call what to do based on error propagated from CoreWorkerPlasmaStoreProvider.